lib/Dialect/FIRRTL/Transforms/CheckCombLoops.cpp

```diff
@@ -421,6 +436,9 @@ class DiscoverLoops {
               continue;
             }
         }
+        if (!type_isa<RefType>(srcArgType) && sinkPortIsForceable)
+          recordProbe(srcPort, sinkPort);
```

Comment:
I believe you but if you could spare a word for what this does/why that'd be swell.  Thank you!

If there's a path from a non-probe to a rwprobe, record a probe edge sink -> source? I'm not sure that's right, consider:

```
FIRRTL version 3.3.0
circuit Bar :
  module Foo :
    input clock : Clock
    output out : Clock
    output clockProbe_bore : RWProbe<Clock>

    node n = clock

    define clockProbe_bore = rwprobe(n)
    connect out, clock

  module Bar :
    input clock : Clock
    output clockProbe : RWProbe<Clock>

    inst foo of Foo
    connect foo.clock, clock
    define clockProbe = foo.clockProbe_bore

    force(clock, UInt<1>(1), clockProbe, foo.out)
```

Which reports a cycle but there isn't one, forcing `n` is unused.

Reply:
Thanks for spotting this, I need to fix the logic to track which ports can be updated by a probe. Working on it.